ProfitApp/src/main/java/bsu/fpmi/profit/ProfitCRUD.java

```diff
@@ -0,0 +1,424 @@
+        List<String> tags = new ArrayList<>();
+
+        String sql = "select * from hashtags where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        while(rs.next()){
+            tags.add(rs.getString("tag"));
+        }
+        rs.close();
+        preparedStatement.close();
+        return tags;
+    }
+
+    private class ReviewsWrapper{
+        private List<String> reviews;
+        private double rating;
+
+        public List<String> getReviews() {
+            return reviews;
+        }
+
+        public double getRating() {
+            return rating;
+        }
+
+        public void setRating(double rating) {
+            this.rating = rating;
+        }
+
+        public void setReviews(List<String> reviews) {
+            this.reviews = reviews;
+        }
+    }
+
+    private ReviewsWrapper readReviews(String id) throws SQLException {
+        List<String> reviews = new ArrayList<>();
+        ReviewsWrapper rw = new ReviewsWrapper();
+        double rating = 0.0;
+        int amountReviews = 0;
+        String sql = "select * from reviews where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        while(rs.next()){
+            rating += rs.getInt("rating");
+            amountReviews++;
+            reviews.add(rs.getString("review"));
+        }
+        if(amountReviews != 0) {
+            rating /= amountReviews;
+        }
+
+        rw.setRating(rating);
+        rw.setReviews(reviews);
+        return rw;
+    }
+
+    private Ad readOffer(String id, List<String> tags, List<String> reviews, double rating) throws SQLException {
+        String sql = "select * from offer " +
+                "left join user on offer.user_id=user.user_id " +
+                "where offer_id = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement.setInt(1, Integer.parseInt(id));
+        ResultSet rs = preparedStatement.executeQuery();
+        if(rs.next()){
+            Ad offer = new Ad(id,
+                    rs.getString("offer_name"),
+                    rs.getString("description"),
+                    rs.getString("created_at"),
+                    rs.getString("vendor_link"),
+                    rs.getString("username"),
+                    rs.getString("photo_link"),
+                    tags,
+                    rs.getString("discount"),
+                    rs.getString("valid_until"),
+                    rating,
+                    reviews);
+            return offer;
+        }
+        else{
+            return null;
+        }
+    }
+
+    public Ad getOffer(String id){
```

Comment:
See what I was speaking of above?) It's too bad the public method `getOffer` calls the private `readOffer`, but the latter is placed above the former